PowerEditor/src/Parameters.h

```diff
@@ -177,6 +177,8 @@ struct sessionFileInfo : public Position
 	std::vector<size_t> _marks;
 	std::vector<size_t> _foldStates;
 	int	_encoding = -1;
+	bool _isUserReadOnly = false;
+	bool _isMonitoring = false;

```

Comment:
Why do we need this member variable?

Reply:
@donho 

we need `_isMonitoring` because we can't rely only on `_isUserReadOnly` as this flag is also set when monitoring is ON.

So to save readonly for a file in session we need to check whether the file is really set as readonly or it is set readonly because of monitoring. This check is being done [here](
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/4987/files#diff-35d82bdf4e1fdc62d667eb514cbc8cd2R3145)